### PR TITLE
Added ability to edit in VSTS and trigger site rebuild from VSTS repo.

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**

### CODE REVIEW

File: src/partials/partial-admin-toolbar.hbs [hunk 1]
```diff
@@ -17,13 +17,19 @@
             </ul>
           {{/ inherit }}
           <ul class="list-inline pull-right">
-            {{# inherit stache.config.editInGitHub editInGitHub }}
-              <li>{{{ include stache.config.partial_edit_github }}}</li>
-            {{/ inherit }}
+            {{# if stache.config.editInVSTS }}
```
Comment (Blackbaud-BobbyEarl) [inline src/partials/partial-admin-toolbar.hbs:20]: Is the `if` block necessary versus the new `inherit` at the same level?  I'm not passionate either way, just wasn't sure if `editInVSTS` and `editInGitHub` are mutually exclusive.

Comment (blackbaud-brandonhare) [reply src/partials/partial-admin-toolbar.hbs:20]: They are mutually exclusive, but I wanted an extra check in there to make sure. In theory, someone could forget to turn off the `editInGitHub` (it's enabled by default) and have both links render. This is just a failsafe.

File: src/helpers/helpers.js [hunk 2]
```diff
@@ -950,6 +950,26 @@ module.exports.register = function (Handlebars, options, params) {
...
+        editInVSTSLink: function (options) {
+            var src = options.hash.src || (typeof this.page !== 'undefined' ? this.page.src : '');
+            return [
+                stache.config.vsts_protocol,
+                stache.config.vsts_base,
+                '/',
+                stache.config.vsts_project,
+                '/_git/',
+                stache.config.vsts_repo,
+                '?path=%2F',
```
Comment (Blackbaud-BobbyEarl) [inline src/helpers/helpers.js]: That `%2F` is an encoded forward slash.  Looking at what VSTS expects, I would remove the leading `%2F` and instead send `src` through the `encodeURIComponent` function.

File: stache.yml [hunk 3]
```diff
@@ -153,6 +155,11 @@ github_token:
 github_org: blackbaud
 github_repo:
 github_branch: master
+vsts_protocol: https://
+vsts_base: blackbaud.visualstudio.com
+vsts_token:
+vsts_project:
+vsts_branch: master
```
Comment (Blackbaud-BobbyEarl) [inline stache.yml]: Even though I know the end-user needs to specify it, for consistency of known config params should we add `vsts_repo` here?

File: src/helpers/helpers.js (hunk 2 above)
Comment (blackbaud-brandonhare) [reply src/helpers/helpers.js]: Ah, good call. I'll update that.

File: stache.yml (hunk 3 above)
Comment (blackbaud-brandonhare) [reply stache.yml]: Should for consistency. I'll add it.

File: src/helpers/helpers.js [hunk 4]
```diff
@@ -950,6 +950,27 @@ module.exports.register = function (Handlebars, options, params) {
...
+            encodeURIComponent(src);
+            return [
+                stache.config.vsts_protocol,
+                stache.config.vsts_base,
+                '/',
+                stache.config.vsts_project,
+                '/_git/',
+                stache.config.vsts_repo,
+                '?path=',
+                src,
```
Comment (Blackbaud-BobbyEarl) [inline src/helpers/helpers.js]: `encodeURIComponent` returns a new string, so you'll need to move line 957 down to here.

Comment (blackbaud-brandonhare) [reply src/helpers/helpers.js]: Hmm. I actually don't need the encoded part in that path for it to work, so I may just remove it entirely.

Comment (Blackbaud-BobbyEarl) [reply src/helpers/helpers.js]: The browser is being nice and handling it for you, but anytime you're passing reserved characters (which forward slash is) around, you should pass it through `encodeURIComponent` first.

**Review by Blackbaud-BobbyEarl — APPROVED** [review]

**Comment by Blackbaud-BobbyEarl** [comment]
Approved

**Pull request merged**